addons/binding/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/rio/AbstractBridgeHandler.java

```diff
@@ -77,6 +98,38 @@ public SocketSession getSocketSession() {
     }
 
     /**
+     * Returns the {@link RioPresetsProtocol} for this {@link Bridge}. The default implementation is to look in the
+     * parent
+     * {@link #getPresetsProtocol()} for the {@link RioPresetsProtocol}
+     *
+     * @return a {@link RioPresetsProtocol} or null if none exists
+     */
+    @SuppressWarnings("rawtypes")
+    public RioPresetsProtocol getPresetsProtocol() {
+        final Bridge bridge = getBridge();
+        if (bridge.getHandler() instanceof AbstractBridgeHandler) {
+            return ((AbstractBridgeHandler) bridge.getHandler()).getPresetsProtocol();
+        }
+        return null;
+    }
+
+    /**
+     * Returns the {@link RioSystemFavoritesProtocol} for this {@link Bridge}. The default implementation is to look in the
+     * parent
+     * {@link #getPresetsProtocol()} for the {@link RioSystemFavoritesProtocol}
+     *
+     * @return a {@link RioSystemFavoritesProtocol} or null if none exists
+     */
+    @SuppressWarnings("rawtypes")
+    public RioSystemFavoritesProtocol getSystemFavoritesHandler() {
+        final Bridge bridge = getBridge();
+        if (bridge != null && bridge.getHandler() instanceof AbstractBridgeHandler) {
```

Comment:
Martin - btw just wanted to thank you for the time reviewing.  I have no idea what you mean by this comment - would you mind explaining a bit more?